AppendProcessorTests.testAppendingToListWithDuplicatesDisallowed can fail with assertion

**Build scan**: https://gradle-enterprise.elastic.co/s/dlgmkmibgzt2a

**Repro line**:

```
./gradlew ':modules:ingest-common:test' --tests "org.elasticsearch.ingest.common.AppendProcessorTests.testAppendingToListWithDuplicatesDisallowed" -Dtests.seed=7B44849061BF4D9F -Dtests.security.manager=true -Dtests.locale=th-TH -Dtests.timezone=America/Argentina/Cordoba -Druntime.java=11
```

**Reproduces locally?**: yes

**Applicable branches**: master

**Failure history**: failed twice recently, once on Sept 15th and once today.

**Failure excerpt**:
```
11:08:02   2> java.lang.AssertionError: 
11:08:02     Expected: iterable with items ["JZonyzmDZ", "qvEz", "TOGLQxvUng", "FOeEwuE", "XUafHwFLfr", "NvW", "Y", "rDU", "XsDDBt", "bzvqy", "Y", "wBMUs"] in any order
11:08:02          but: no item matches: "Y" in ["JZonyzmDZ", "qvEz", "TOGLQxvUng", "FOeEwuE", "XUafHwFLfr", "NvW", "Y", "XsDDBt", "bzvqy", "rDU", "wBMUs"]
11:08:02         at __randomizedtesting.SeedInfo.seed([7B44849061BF4D9F:699BB024BAB4EF35]:0)
11:08:02         at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)
11:08:02         at org.junit.Assert.assertThat(Assert.java:956)
11:08:02         at org.junit.Assert.assertThat(Assert.java:923)
11:08:02         at org.elasticsearch.ingest.common.AppendProcessorTests.testAppendingToListWithDuplicatesDisallowed(AppendProcessorTests.java:222)
```

--- a/modules/ingest-common/src/test/java/org/elasticsearch/ingest/common/AppendProcessorTests.java
+++ b/modules/ingest-common/src/test/java/org/elasticsearch/ingest/common/AppendProcessorTests.java
@@ -193,7 +193,6 @@ public class AppendProcessorTests extends ESTestCase {
         assertThat(list, equalTo(List.of(originalValue, newValue)));
     }
 
-    @AwaitsFix(bugUrl = "https://github.com/elastic/elasticsearch/issues/62839")
     public void testAppendingToListWithDuplicatesDisallowed() throws Exception {
         IngestDocument ingestDocument = RandomDocumentPicks.randomIngestDocument(random());
         int size = randomIntBetween(0, 10);
@@ -209,7 +208,7 @@ public class AppendProcessorTests extends ESTestCase {
             .collect(Collectors.toSet());
 
         // create a set using the new values making sure there are no overlapping values already present in the existing values
-        Set<String> nonexistingValues = Sets.difference(newValues, new HashSet<>(existingValues));
+        Set<String> nonexistingValues = Sets.difference(newValues, new HashSet<>(list));
         List<String> valuesToAppend = new ArrayList<>(existingValues);
         valuesToAppend.addAll(nonexistingValues);
         expectedValues.addAll(nonexistingValues);

